flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/common/RankTestPrograms.java

```diff
@@ -184,4 +207,32 @@ private static TableTestProgram getTableTestProgram(
                                     + "(select a, b, row_number() over(partition by a order by t asc) as c from MyTable1)"
                                     + " where c <= 2")
                     .build();
+
+    public static final TableTestProgram RANK_2_TEST =
```

Comment:
As a note, this test is not supported in batch execution mode.  An error `Table RANK() on streaming table is not supported currently` is returned.